File: medplum-java/medplum-model/src/main/java/com/fhir/Provenance_Entity.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableProvenance_Entity.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface Provenance_Entity {
  Optional<List<Provenance_Agent>> agent();

  Optional<String> id();

  Optional<List<Extension>> extension();

  Optional<List<Extension>> modifierExtension();

  Optional<Provenance_entityRole> role();

  Reference what();

  static ImmutableProvenance_Entity.WhatBuildStage builder() {
    return ImmutableProvenance_Entity.builder();
  }
}

File: medplum-typescript/src/model/fhir/Provenance_Entity.ts
import { Extension } from "./Extension";
import { Provenance_Agent } from "./Provenance_Agent";
import { Provenance_entityRole } from "./Provenance_entityRole";
import { Reference } from "./Reference";

export interface Provenance_Entity {
    id: string | null | undefined;
    extension: Extension[] | null | undefined;
    modifierExtension: Extension[] | null | undefined;
    role: Provenance_entityRole | null | undefined;
    what: Reference;
    agent: Provenance_Agent[] | null | undefined;
}
